lib/rules/newline-after-var.js

```diff
@@ -35,6 +35,29 @@ module.exports = function(context) {
     }
 
     /**
+     * Return the line number on which the given `token` starts
+     * @private
+     * @param {ASTNode} token - token to query
+     * @returns {Number} line number
+     */
+    function getLineNum(token) {
+        return token.loc.start.line;
+    }
+
+    /**
+     * Check if a comment exists on the given line number
+     * @private
+     * @param {Number} lineNum - line number to test
+     * @returns {Boolean} `true` if line contains a comment
+     */
+    function isCommentLine(lineNum) {
```

Comment:
This is pretty expensive to call for every variable declaration. Can you cache the mapped comments?